operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/UpdateControl.java

```diff
@@ -82,11 +85,31 @@ public static <T extends HasMetadata> UpdateControl<T> updateResourceAndStatus(
     return new UpdateControl<>(customResource, true, true, false);
   }
 
-  public static <T extends HasMetadata> UpdateControl<T> patchResourceAndStatus(
+  /**
+   * Updates the resource - with optimistic locking - and patched the status without optimistic
+   * locking in place.
+   *
+   * @param customResource
+   * @return
+   * @param <T>
+   */
+  public static <T extends HasMetadata> UpdateControl<T> updateResourceAndPatchStatus(
       T customResource) {
     return new UpdateControl<>(customResource, true, true, true);
   }
 
+  /**
+   * Market for removal, because of confusing name. It does not patch the resource just updates it.
+   * This method is same as updateResourceAndPatchStatus.
+   *
+   * @param customResource to update
+   * @return UpdateControl instance
+   * @param <T> resource type
```

Comment:
Add `@deprecated` with link to method to use instead.

Reply:
added